Android/app/src/main/java/com/example/kahye/common/ReservationActivity.java

```diff
@@ -28,49 +29,57 @@
     ImageButton upButton;
     ImageButton downButton;
     Button alertButton;
+    String selectedDate;
+    Class selectedClass;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_reservation);
+        Bundle bundle = this.getIntent().getExtras();
+
+        // classInfo
+        selectedClass = bundle.getParcelable("_classInfo");
+        setContentView(R.layout.activity_reservation);
 
         // class Img
-        Bundle bundle = this.getIntent().getExtras();
         int pic = bundle.getInt("classImg");
         ImageView classImgView = (ImageView) findViewById(R.id.classImgView);
         classImgView.setImageResource(pic);
 
-        final String getTime = getIntent().getStringExtra("Date");
+        selectedDate = bundle.getString("_date");
         TextView dateView = (TextView) findViewById(R.id.dateView);
-        dateView.setText(getTime);
+        dateView.setText(selectedDate);
 
         // time list
         ListView timeListView = (ListView) findViewById(R.id.timeListView);
         final List<String> timeList = new ArrayList<>();
         ArrayAdapter<String> adapter = new ArrayAdapter<>(this,
                 android.R.layout.simple_list_item_1, timeList);
 
-        //TODO (gayeon) : load data from server
-        timeList.add("2:00PM ~ 4:00PM");
-        timeList.add("4:00PM ~ 6:00PM");
-        timeList.add("6:00PM ~ 8:00PM");
-        timeList.add("8:00PM ~ 10:00PM");
+        List<timeTable>  timeslot = selectedClass.getAvailableTimeTable();
+        for (int timeListIdx = 0; timeListIdx < timeslot.size(); timeListIdx++){
+            String timeString = timeslot.get(timeListIdx).getStartTime().toString() + " ~ " +
```

Comment:
I would have to look at the demo, but is this the right place to construct UI elements? This is fine for now, but would have to be fixed if we want to internationalize this code (since some languages are displayed/read right-to-left).

Reply:
Oh, I have not to think about this problem yet.
We need to talk about it later.